docs/src/content/docs/datamessages.mdx

```diff
@@ -0,0 +1,88 @@
+| [InputTextMessage](#inputtextmessage)         | Client | Used to send a user message to the agent via text.   |
+| [SetOutputMedium](#setoutputmedium)           | Client | Sets server's output medium to text or voice.        |
+| [ClientToolInvocation](#clienttoolinvocation) | Server | Asks the client to invoke a client tool.             |
+| [ClientToolResult](#clienttoolresult)         | Client | Contains the result of a client tool invocation.     |
+| [Debug](#debug)                               | Server | Useful for application debugging.                    |
+| [PlaybackClearBuffer](#playbackclearbuffer)   | Server | Used to clear buffered output audio. WebSocket only. |
+
+
+## Ping
+A message sent by the client to measure round-trip data message latency.
+- `type: "ping"` 
+- `timestamp`: Float. Client timestamp for latency measurement.
+
+## Pong
+A message sent by the server in response to a PingMessage. The timestamp is copied from the PingMessage.
+- `type: "pong"`
+- `timestamp`: Float. Matching ping timestamp.
+
+## State
+A message sent by the server to indicate its current state.
+- `type: "state"`
+- `state`: Current session state
+
+## Transcript
+A message containing text transcripts of user and agent audio utterances.
+- `type: "transcript"`
+- `role`: "user" or "agent". Who emitted the utterance.
+- `medium`: "text" or "voice". The medium through which the utterance was emitted.
+- `text`: String. Full transcript text (exclusive with delta). The full text of the transcript so far. Either this or delta will be set.
+- `delta`: String. Incremental transcript update (exclusive with text). The additional transcript text added since the last agent transcript message.
+- `final`: Boolean. Whether more updates expected. Whether to expect additional transcript messages for this conversation round.
+
+## InputTextMessage
+A user message sent via text.
+- `type: "input_text_message"`
+- `text`: String. The content of the user message.
+
+## SetOutputMedium
+Message sent by the client to set the server's output medium.
+- `type: "set_output_medium"`
+- `medium`: Either "voice" or "text".
+
+## ClientToolInvocation
+Sent by the server to ask the client to invoke a client-implemented tool with the given parameters. The client is expected to send back a ClientToolResultMessage with a matching invocation_id.
+- `type: "client_tool_invocation"`
+- `tool_name`: String. Tool to invoke
+- `invocation_id`: String. Unique invocation ID
+- `parameters`: Dict[String, Any]. Tool parameters
+
+## ClientToolResult
+Contains the result of a client-implemented tool invocation.
+- `type: "client_tool_result"`
+- `invocation_id`: String. Matches corresponding invocation.
+- `result`: String. Tool execution result (optional).
+- `response_type`: String. Defaults to "tool-response".
+- `error_type`: "undefined" or "implementation-error" (optional).
```

Comment:
```suggestion
- `error_type`: Optional string. Should be omitted when result is set. Otherwise, should be "undefined" if the a tool with the given name does not exist or "implementation-error" otherwise.
```